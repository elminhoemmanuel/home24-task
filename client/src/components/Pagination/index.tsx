import React from 'react'

type PaginationProps = {
    totalArticles: number;
    range: number;
    viewMore: () => void;
    viewLess: () => void;
}

const Pagination = ({ totalArticles, range, viewMore, viewLess }:PaginationProps) => {
    return (
        <div className='flex items-center justify-center py-6'>
            {
                range >= totalArticles ?
                    <button data-testid="less-btn" onClick={() => viewLess()} className='block text-hmred'>
                        See Less
                    </button>
                    :
                    <button data-testid="more-btn" onClick={() => viewMore()} className='block text-hmred'>
                        See More
                    </button>
            }
        </div>
    )
}

export default Pagination