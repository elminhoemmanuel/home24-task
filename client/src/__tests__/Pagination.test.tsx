import React from 'react';
import { render, screen } from '@testing-library/react';
import Pagination from '../components/Pagination';


test("On render, check that Pagination see more button is rendered correctly", () => {
    const range = 20;
    const totalArticles = 50;
    const viewMore = (): void => { };
    const viewLess = (): void => { };

    const { getByTestId } = render(<Pagination range={range} totalArticles={totalArticles} viewLess={viewLess} viewMore={viewMore} />);

    if (range < totalArticles) {
        expect(screen.getByTestId("more-btn"))
    }
});

test("On render, check that Pagination see less button is rendered correctly", () => {
    const range = 60;
    const totalArticles = 50;
    const viewMore = (): void => { };
    const viewLess = (): void => { };

    const { getByTestId } = render(<Pagination range={range} totalArticles={totalArticles} viewLess={viewLess} viewMore={viewMore} />);

    if (range >= totalArticles) {
        expect(screen.getByTestId("less-btn"))
    }
});
