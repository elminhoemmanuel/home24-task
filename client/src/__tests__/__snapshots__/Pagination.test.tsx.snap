// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On render, check that Sidebar list is rendered with correctly 1`] = `
<div
  class="flex items-center justify-center py-6"
>
  <button
    class="block text-hmred"
  >
    See More
  </button>
</div>
`;
